### ISSUE TITLE
Hierarchical Category URLs

### ISSUE THREAD

**Issue by shopp-lighthouse-import** [ingenesis/shopp#1811]
*Imported from Lighthouse* http://shopp.lighthouseapp.com/projects/47561/tickets/1811
Reported by **Jonathan** at **Wednesday, 23-May-12 11:05:07 GMT-0400**

Originally this ticket was posted by Brandon Turner and included as part of a set of patches for both hierarchical breadcrumbs and category URLs in ticket #1711:

> Hierarchical categories were supported in 1.1. In 1.2, URLs and breadcrumbs no longer support hierarchical categories. Categories appear flat. It would be nice if we could continue using hierarchical urls and breadcrumbs in 1.2.

We accepted the patches for breadcrumbs, but not the hierarchical category URLs because of the potential for breakage it could cause throughout the system. An audit needs run to funnel all category/collection URL use internally when generating URLs for categories or smart collections to use the `shopp('collection','url')` tag.

Additionally we need to consider adding hierarchical category URL support as an option in Shopp settings. One idea suggested by John Dillick is to change the **Pages** menu under **Setup** to **Permalinks** so that it can cover a broader spectrum of URL-related settings.

**Comment by WazzaJB** [comment]
Just thought i'd add a demonstration of what this PR will effectively allow developers to achieve (as I am currently using it).

Currently the category structure is:

`shop/category/category1child/`

With this code:

```
add_filter('shopp_category_rewrite_args', 'modify_shopp_category_rewrite');
function modify_shopp_category_rewrite( $rewrite )
{
	return array( 'slug' => ShoppPages()->baseslug(), 'with_front' => false, 'hierarchical' => true );
}
```

I now have:

`shop/category1/category1child/`

Which strips the 'category' base and allows for hierarchical URLs, so far working flawlessly.

**Comment by jonathandavis** [comment]
Closed with merge of #3185
